### ISSUE TITLE
build failed for suzuran

### ISSUE THREAD

**Issue by oupala** [lineageos4microg/docker-lineage-cicd#312]
I'm trying to build a lineageos for `suzuran`:

```log
Default revision lineage-18.1 not found in android_device_sony_suzuran. Bailing.
Branches found:
cm-14.1
Use the ROOMSERVICE_BRANCHES environment variable to specify a list of fallback branches.
14:28:35 Build sandboxing disabled due to nsjail error.
In file included from build/make/core/config.mk:291:
In file included from build/make/core/envsetup.mk:266:
^[[1mbuild/make/core/product_config.mk:155: ^[[31merror: ^[[0m^[[1mCan not locate config makefile for product "lineage_suzuran".^[[0m
14:28:35 dumpvars failed with: exit status 1
14:28:36 Build sandboxing disabled due to nsjail error.
In file included from build/make/core/config.mk:291:
In file included from build/make/core/envsetup.mk:266:
^[[1mbuild/make/core/product_config.mk:155: ^[[31merror: ^[[0m^[[1mCan not locate config makefile for product "lineage_suzuran".^[[0m
14:28:36 dumpvars failed with: exit status 1

** Don't have a product spec for: 'lineage_suzuran'
** Do you have the right repo manifest?

>> [Sat Jun 18 14:28:36 UTC 2022] breakfast failed for suzuran, lineage-18.1 branch
```

How can I find or create the manifest for this suzuran device ([Sony Xperia Z5 Compact](https://doc.e.foundation/devices/suzuran))?

**Comment by ElCorzo** [comment]
According to this page:
https://github.com/LineageOS/lineage_wiki/blob/master/_data/devices/suzuran.yml
suzuran is supported on Version 14.1 and according to this page
https://wiki.lineageos.org/devices/suzuran/
"no longer maintained".
I'm afraid you have to port yourself. Can you build a 14.1 image?

**Comment by oupala** [comment]
I'm afraid *docker-lineage-cicd* does not currently support LineageOS version 14.1:

> [Mon Jul  4 22:54:18 UTC 2022] Building branch lineage-14.1 is not (yet) suppported

**Comment by ElCorzo** [comment]
According to https://github.com/LineageOS/android/branches it is called cm-7.0.1 (*not* cm-14.1).
Maybe it works ...

**Comment by se3now** [comment]
Hi, I also have this problem with my unsupported old on5xelte phone.

I used this command for building:
```
docker run \
    -e "BRANCH_NAME=cm-14.1" \
    -e "DEVICE_LIST=on5xelte" \
    -e "SIGN_BUILDS=true" \
    -e "SIGNATURE_SPOOFING=restricted" \
    -e "WITH_GMS=true" \
    -e "INCLUDE_PROPRIETARY=false" \
    -v "/srv/e/src:/srv/src" \
    -v "/srv/e/zips:/srv/zips" \
    -v "/srv/e/logs:/srv/logs" \
    -v "/srv/e/ccache:/srv/ccache" \
    -v "/srv/e/keys:/srv/keys" \
    -v "/srv/e/local_manifests:/srv/local_manifests" \
    lineageos4microg/docker-lineage-cicd
```

and I got this error:
```
including vendor/cm/vendorsetup.sh
build/core/product_config.mk:249: *** Can not locate config makefile for product "lineage_on5xelte".  Stop.
build/core/product_config.mk:249: *** Can not locate config makefile for product "cm_on5xelte".  Stop.
build/core/product_config.mk:249: *** Can not locate config makefile for product "lineage_on5xelte".  Stop.
Device on5xelte not found. Attempting to retrieve device repository from LineageOS Github (http://github.com/LineageOS).
Repository for on5xelte not found in the LineageOS Github repository list. If this is in error, you may need to manually add it to your local_manifests/roomservice.xml.
build/core/product_config.mk:249: *** Can not locate config makefile for product "lineage_on5xelte".  Stop.
build/core/product_config.mk:249: *** Can not locate config makefile for product "lineage_on5xelte".  Stop.

** Don't have a product spec for: 'lineage_on5xelte'
** Do you have the right repo manifest?

>> [Mon Aug 22 08:51:24 UTC 2022] breakfast failed for on5xelte, cm-14.1 branch
```

and this is my custom manifest in /srv/e/local_manifests

```
<?xml version="1.0" encoding="UTF-8"?>
<manifest>
    <project name="se3now/android_device_samsung_on5xelte" path="device/samsung/on5xelte" remote="github" revision="cm-14.1" />
    <project name="se3now/android_kernel_samsung_on5xelte" path="kernel/samsung/on5xelte" remote="github" revision="cm-14.1" />
    <project name="TheMuppets/proprietary_vendor_samsung" path="vendor/samsung" remote="github" revision="cm-14.1" />
    <project name="LineageOS/android_device_qcom_common" path="device/qcom/common" remote="github" revision="cm-14.1" />
</manifest>

```

**Comment by petefoth** [comment]
> How can I find or create the manifest for this suzuran device ([Sony Xperia Z5 Compact](https://doc.e.foundation/devices/suzuran))?

There is currently a well-supported and maintained unofficial build of LineageOS for this device. The `19.1` branch is pretty stable, and still getting regular updates: details [in this XDA thread](https://forum.xda-developers.com/t/rom-unofficial-12-s-lineageos-19-1-for-z5c-suzuran.4417131/). The build includes F-Droid, Aurora Store and UnifieldNLP location module, but not MicroG GmsCore.

The same developer (Bernhard Thoben, Professor-Berni  on github) has a 20.0 build as well, which is currently only Alpha quality - see [here](https://forum.xda-developers.com/t/rom-unofficial-alpha-13-t-lineageos-20-0-for-z5c-suzuran.4508617/#post-87619455)

All the necessary sources and vendor blobs are available [in his repos](https://github.com/Professor-Berni) along with needed manifests, patches and init scripts.

Building a lineageos4microg is not straightforward using this Docker image, as various init scripts need to be run and various patches need to be applied, some of which seem to conflict with the signature spoofing patches from here.

I have managed to make unofficial and custom 19.1 lineage4microg builds, which can be found on [AndroidFileHost](https://www.androidfilehost.com/?w=files&flid=322410).  More information about these builds can be found at https://codeberg.org/petefoths-projects/unofficial-l4m-builds.

**Comment by petefoth** [comment]
In theory it is still possible to make CM14 builds with our docker image, but we don't know if anyone is actually ding so successfully. We don;t really have time to investigate problems with building branches we no longer build for ourselves (i.e. 17.1 and earlier.

As mentioned above @Professor-Berni is currently making up-to-date builds (18.1, 19.1 & 20.1) for this device, and his builds now support / include microG, and pretty much everything else that los4mg has.

Unfortunately, he makes his builds using the [`LineageOS Ultra Legacy' project sources](https://github.com/LineageOS-UL/android) rather than the LineageOS sources we use. That makes making unofficial builds of los4mg quite difficult: I spent a couple of days last week trying, but I decided t give up

If you want to carry on using this excellent Sony device, then I strongly recommend trying out one of his builds. Mine is crrently running the latest 19.1 build from [here](https://forum.xda-developers.com/t/rom-unofficial-12-s-lineageos-19-1-for-z5c-suzuran.4417131/). The 20.0 still has too many rough edges for me.

As far as this issue is concerned feel free to either leave it open, though we won't be able to spend any time on it, or to close it.

Assigning back to @oupala for a response
